docker/openemr/flex-edge/autoconfig.sh

```diff
@@ -144,13 +146,14 @@ if [ -f /etc/docker-leader ] ||
            [ "$SWARM_MODE" == "yes" ]; then
             touch openemr/sites/default/docker-initiated
         fi
-        rsync --recursive --links --exclude .git openemr /var/www/localhost/htdocs/
+        rsync --ignore-existing --recursive --links --exclude .git openemr /var/www/localhost/htdocs/
         rm -fr openemr
         cd /var/www/localhost/htdocs/
     fi
 
     if [ -f /var/www/localhost/htdocs/auto_configure.php ] &&
-       [ ! -d /var/www/localhost/htdocs/openemr/vendor ] &&
+       [ -d /var/www/localhost/htdocs/openemr/vendor ] &&
+       [ -z "$(ls -A /var/www/localhost/htdocs/openemr/vendor)" ]  &&
```

Comment:
What are the above 2 lines meant to do?

Reply:
"if the vendor directory exist and it is empty"
When i was mounting the vendor directory, it mounted an empty dir. so it was skipping these build steps.

It might be better if we say "if the vendor directory doesn't exist or if the vendor directory is empty" but I couldn't figure out how to write that in bash...

Reply:
But won't this break is a user is using this and has an empty vendor directory?
If so, then need to ensure it works for both cases:
1. A nonexistent vendor directory
2. An empty vendor directory

Reply:
regarding above, meant to say:
But won't this break is a user is using this and has no vendor directory?

Reply:
Yeah... If you can craft an if statement that'll check both cases, I'd be grateful. I wasn't able to figure it out when I was working on it.

Reply:
try:
```
    if [ -f /var/www/localhost/htdocs/auto_configure.php ] &&
       [[ ! -d /var/www/localhost/htdocs/openemr/vendor || ( -d /var/www/localhost/htdocs/openemr/vendor  && -z "$(ls -A /var/www/localhost/htdocs/openemr/vendor)" ) ]]  &&
       [ "$FORCE_NO_BUILD_MODE" != "yes" ]; then
```
I have no idea if it will work, but adapted it from here: https://stackoverflow.com/questions/14964805/groups-of-compound-conditions-in-bash-test

Reply:
https://stackoverflow.com/questions/3427872/whats-the-difference-between-and-in-bash for a little more insight into what's going on there.

Reply:
The above should work (fingers crossed) :)

Reply:
Close! Had to escape the parens:
```bash
    if [ -f /var/www/localhost/htdocs/auto_configure.php ] &&
       [[ ! -d /var/www/localhost/htdocs/openemr/vendor || \( -d /var/www/localhost/htdocs/openemr/vendor &&  -z "$(ls -A /var/www/localhost/htdocs/openemr/vendor)" \) ]]  &&
       [ "$FORCE_NO_BUILD_MODE" != "yes" ]; then
```

See: https://stackoverflow.com/a/3826462/3164104